code/modules/mob/living/simple_animal/abnormality/aleph/army_in_black.dm

```diff
@@ -0,0 +1,498 @@
+/mob/living/simple_animal/hostile/army_friend/AttackingTarget()
+	OpenFire()
+	return FALSE
+
+/mob/living/simple_animal/hostile/army_friend/OpenFire()
+	if(protection_cooldown <= world.time && prob(75) && !client)
+		FriendlyHeal()
+		return
+
+/mob/living/simple_animal/hostile/army_friend/PickTarget(list/Targets)//don't de-aggro your target unless they're dead
+	if(protection_target)
+		Targets = list()
+		Targets += protection_target
+	..()
+
+//friendly radial heal
+/mob/living/simple_animal/hostile/army_friend/proc/FriendlyHeal()
+	if(stat == DEAD)
+		return
+	for(var/turf/T in view(3, src))
+		new /obj/effect/temp_visual/friend_hearts(get_turf(T))
+		for(var/mob/living/carbon/human/H in T.contents)
+			H.adjustBruteLoss(-6)
+			H.adjustSanityLoss(-6)
+	playsound(get_turf(src), 'sound/abnormalities/armyinblack/pink_heal.ogg', 50, 0, 2)
+
+//friendly explosion
+/mob/living/simple_animal/hostile/army_friend/proc/PinkExplode()
+	if(QDELETED(src))
+		return
+	playsound(get_turf(src), 'sound/abnormalities/armyinblack/pink_explosion.ogg', 125, 0, 8)
+	for(var/mob/living/carbon/human/H in view(7, src))
+		H.adjustBruteLoss(-20)
+		H.adjustSanityLoss(20)
+	new /obj/effect/temp_visual/pink_explosion(get_turf(src))
+	qdel(src)
+
+//hostile breach mob
+/mob/living/simple_animal/hostile/army_enemy
+	name = "Army In Black"
+	desc = "Yes.. we, the Army in Black.. blend into the human heart.. and drive away good thoughts.."
+	icon = 'ModularTegustation/Teguicons/64x64.dmi'
+	icon_state = "armyinblack"
+	icon_living = "armyinblack"
+	icon_dead = "armyinblack_heart"
+	pixel_x = -16
+	base_pixel_x = -16
+	faction = list("hostile")
+	attack_verb_continuous = "attacks"
+	attack_verb_simple = "attack"
+	attack_sound = 'sound/abnormalities/armyinblack/pink_heal.ogg'
+	/*Core stats*/
+	health = 900
+	maxHealth = 900
+	obj_damage = 50
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1.2, WHITE_DAMAGE = 0.6, BLACK_DAMAGE = 1.0, PALE_DAMAGE = 0.8)
+	ranged = TRUE
+	minimum_distance = 2
+	speed = 2
+	move_to_delay = 20//TODO: make this determined by how far away it spawned from the beacon
+	stat_attack = HARD_CRIT
+	del_on_death = FALSE
+	density = FALSE
+	is_flying_animal = TRUE
+	var/list/fear_affected = list()
+	var/can_act = TRUE
+	var/shot_cooldown
+	var/shot_cooldown_time = 5 SECONDS
+	var/boom_damage = 70
+	var/boom_radius = 20
+	var/targetted_beacon
+	var/list/moving_path
+
+//movement and AI
+/mob/living/simple_animal/hostile/army_enemy/AttackingTarget()
+	return
+
+/mob/living/simple_animal/hostile/army_enemy/proc/GoToBeacon()
+	if(QDELETED(src))
+		return
+	patrol_to(get_turf(targetted_beacon))//this causes a runtime when the unit is deleted due to invoking asynchronous move_to! Please help!
+	return
+
+/mob/living/simple_animal/hostile/army_enemy/Life()
+	. = ..()
+	if(!.)
+		return
+	if(targetted_beacon)
+		GoToBeacon()
+	return
+
+/mob/living/simple_animal/hostile/army_enemy/Initialize()
+	..()
+	addtimer(CALLBACK(src, .proc/Explode), 120 SECONDS)
```

Comment:
So they explode in 2 minutes instead of when they reach their destination?

Reply:
They explode in 2 minutes if they either don't reach their destination in time (They shouldn't) or for whatever reason no destination exists, mainly the latter.